subprojects/platform-jvm/src/main/java/org/gradle/api/java/archives/internal/DefaultAttributes.java

```diff
@@ -38,6 +67,9 @@ public boolean isEmpty() {
 
     @Override
     public boolean containsKey(Object key) {
+        if (key instanceof String) {
+            return attributes.containsKey(keyFor((String) key));
+        }
```

Comment:
❌ The addition of these checks may actually cause unexpected behaviour if a user would use, previously, invalid keys. They would now be checked by the constructor of `Attributes.Name` which may throw an exception. We will need to consider this either via deprecation warning or accepting the potential breaking change. As I mentioned previously, if it was fundamentally broken before, then we could easily accept the breaking change.

Reply:
manifest attributes have to be case-insensitive.
Previous implementation did not consider that, so users could get "not found" response when they seek for the key in "wrong casing".

New implementation:
1) Supports case-insensitive search
2) It fails if input string does not conform to the manifest spec. Well, it did not fail previously (it just returned "not found" response), however, I don't think it is bad to throw exceptions on invalid input keys.

I'm fine to just ignore the exceptions though, however, I think no-one would really use `containsKey` API. The key feature here is "enable case-insensitive search".